C and C++ compilers are not link compatible.  Can not continue.

When running an OpenMPI build (from 284 onwards) we get

# Identification

```
checking if C and C++ are link compatible... no
**********************************************************************
* It appears that your C++ compiler is unable to link against object
* files created by your C compiler.  This generally indicates either
* a conflict between the options specified in CFLAGS and CXXFLAGS
* or a problem with the local compiler installation.  More
* information (including exactly what command was given to the
* compilers and what error resulted when the commands were executed) is
* available in the config.log file in this directory.
**********************************************************************
configure: error: C and C++ compilers are not link compatible.  Can not continue.
```

# Diagnosis

It seems that the compiler is only compiled to create x86 executables, not x86_64 executables, according to http://www.open-mpi.org/community/lists/users/2008/04/5497.php, which claims : 

>  if the gcc output needs to be x86_64 add the -m64 flag to the c and c++ flags.

--- a/build.sh
+++ b/build.sh
@@ -40,7 +40,7 @@ else
 fi
 cd $WORKSPACE/$NAME-$VERSION
 
-./configure --prefix=$SOFT_DIR --with-gmp=$GMP_DIR --with-mpfr=$MPFR_DIR \
+CFLAGS=-m64 FFLAGS=-m64./ configure --prefix=$SOFT_DIR --with-gmp=$GMP_DIR --with-mpfr=$MPFR_DIR \
 --with-mpc=$MPC_DIR --enable-languages=c,c++,fortran,java --disable-multilib
 
 make

